src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx

```diff
@@ -0,0 +1,307 @@
+import { AllCommentsTabContainer_story } from "coral-stream/__generated__/AllCommentsTabContainer_story.graphql";
+import { AllCommentsTabContainer_viewer } from "coral-stream/__generated__/AllCommentsTabContainer_viewer.graphql";
+import { COMMENT_SORT } from "coral-stream/__generated__/AllCommentsTabContainerPaginationQuery.graphql";
+import { AllCommentsTabVirtualizedCommentsLocal } from "coral-stream/__generated__/AllCommentsTabVirtualizedCommentsLocal.graphql";
+
+import AllCommentsTabCommentContainer from "./AllCommentsTabCommentContainer";
+
+interface Props {
+  settings: AllCommentsTabContainer_settings;
+  viewer: AllCommentsTabContainer_viewer | null;
+  story: AllCommentsTabContainer_story;
+  loadMoreAndEmit: () => Promise<any>;
+  hasMore: boolean;
+  isLoadingMore: boolean;
+  currentScrollRef: any;
+  alternateOldestViewEnabled: boolean;
+  commentsOrderBy: COMMENT_SORT;
+  comments: AllCommentsTabContainer_story["comments"]["edges"];
+  newCommentsLength: number;
+}
+
+// Virtuoso settings
+const overscan = 50;
+const increaseViewportBy = 2000;
+const virtuosoHeight = 600;
+
+const AllCommentsTabVirtualizedComments: FunctionComponent<Props> = ({
+  story,
+  settings,
+  viewer,
+  loadMoreAndEmit,
+  hasMore,
+  isLoadingMore,
+  currentScrollRef,
+  alternateOldestViewEnabled,
+  commentsOrderBy,
+  comments,
+  newCommentsLength,
+}) => {
+  const [local, setLocal] = useLocal<
+    AllCommentsTabVirtualizedCommentsLocal
+  >(graphql`
+    fragment AllCommentsTabVirtualizedCommentsLocal on Local {
+      showLoadAllCommentsButton
+      oldestFirstNewCommentsToShow
+      totalCommentsLength
+      zKeyClickedLoadAll
+      addACommentButtonClicked
+      keyboardShortcutsConfig {
+        key
+        source
+        reverse
+      }
+    }
+  `);
+
+  // We need to know if the Load all button has been clicked to help determine whether
+  // to display the Load all button or not.
+  const [
+    loadAllButtonHasBeenClicked,
+    setLoadAllButtonHasBeenClicked,
+  ] = useState(false);
+
+  // Initial comments are used to determine whether the Load all comments button should be
+  // displayed or not.
+  const [initialComments, setInitialComments] = useState<null | {
+    length: number;
+    hasMore: boolean;
+  }>(null);
+
+  useEffect(() => {
+    setLocal({ totalCommentsLength: comments.length });
+  }, [comments.length, setLocal]);
+
+  // This determines if there are more comments to display than the initial 20.
+  // It also takes into account the initial comments loaded since if we start
+  // with fewer than 20, we will never want to display load all as new comments
+  // come in.
+  const moreCommentsForLoadAll = useMemo(() => {
+    return (
+      comments.length > NUM_INITIAL_COMMENTS &&
+      ((initialComments && initialComments.length > NUM_INITIAL_COMMENTS) ||
+        (initialComments &&
+          initialComments.length === NUM_INITIAL_COMMENTS &&
+          initialComments.hasMore))
```

Comment:
This is just a thought, not necessary for you to do. When I see complex logic like this, I tend to try and word it out via naming like so:

```
const haveMoreThanInitialComments = comments.length > NUM_INITIAL_COMMENTS;

const initialCommentsMoreThanExpected = initialComments && intialComments.length > NUM_INITIAL_COMMENTS;

const initialCommentsHasMore = initialComments &&
   initialComments.length === NUM_INITIAL_COMMENTS &&
   initialComments.hasMore;
   
return haveMoreThanInitialComments && (initialCommentsMoreThanExpected || initialCommentsHasMore);
```

Not saying this is a good example, but I know that the compiler/minifier can optimize this for me, so I tend to try and name/group logic together based on what my human brain was thinking when writing sections of it.

I suppose `moreCommentsForLoadAll` explains this, but each of the edge cases for why there are "more" is sometimes confusing without more explanation. On further thought, the comments may describe this thoroughly enough already.

Just curious what we think of naming logic bits for clarity.

Reply:
I agree that especially in this case, better naming/grouping of logic here will improve the readability without relying on the comment to explain as much. I was also able to update to simplify the logic with the updates to the way that we are loading in comments now!